src/v3.rs

```diff
@@ -339,11 +340,45 @@ impl Personalization {
 
     /// Add a dynamic template data field.
     pub fn add_dynamic_template_data(mut self, dynamic_template_data: SGMap) -> Personalization {
-        self.dynamic_template_data
-            .get_or_insert_with(|| SGMap::with_capacity(dynamic_template_data.len()))
-            .extend(dynamic_template_data);
+        // We can safely unwrap & unreachable here since SGMap will always serialize
+        // to a JSON object.
+        let mut new_vals = match to_value(dynamic_template_data).unwrap() {
```

Comment:
If this will _always_ succeed, can we drop the `match` statement?

Reply:
excuse my naïveté, however I don't believe that you can select an enum variant without a match or if let:
https://stackoverflow.com/questions/45473656/read-from-an-enum-without-pattern-matching

we can do an if let but that seems less clean since there wouldn't really be branching logic